Duration unit cannot be a negative number

## Bug report 

(Please follow this template replacing the text between parentheses with the requested information)

### Expected behavior and actual behavior
The expected behavior occurred. i.e. pipleine ran all steps and `publishDir` transferred all files.  This error occurred at the end of processing for unknown reasons.
Actual behavior:
```
Nov-19 09:59:56.205 [main] DEBUG nextflow.Session - Session await > all process finished
Nov-19 09:59:56.282 [main] DEBUG nextflow.Session - Session await > all barriers passed
Nov-19 10:00:01.317 [main] INFO  nextflow.Session - Waiting files transfer to complete (1 files)
Nov-19 10:01:01.321 [main] DEBUG nextflow.Session - Waiting files transfer to complete (1 files)
Nov-19 10:02:01.327 [main] DEBUG nextflow.Session - Waiting files transfer to complete (1 files)
Nov-19 10:02:31.675 [main] DEBUG nextflow.Session - Session aborted -- Cause: Duration unit cannot be a negative number. Expression: (duration >= 0). Values: duration = -1574027111516
Nov-19 10:02:31.822 [main] ERROR nextflow.cli.Launcher - @unknown
java.lang.AssertionError: Duration unit cannot be a negative number. Expression: (duration >= 0). Values: duration = -1574027111516
        at org.codehaus.groovy.runtime.InvokerHelper.assertFailed(InvokerHelper.java:417)
        at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.assertFailed(ScriptBytecodeAdapter.java:670)
        at nextflow.util.Duration.<init>(Duration.groovy:71)
        at nextflow.trace.WorkflowStats.getSucceedDuration(WorkflowStats.groovy:111)
        at nextflow.trace.WorkflowStats.toString(WorkflowStats.groovy:225)
        at org.codehaus.groovy.runtime.InvokerHelper.format(InvokerHelper.java:659)
        at org.codehaus.groovy.runtime.InvokerHelper.toString(InvokerHelper.java:148)
        at org.codehaus.groovy.runtime.InvokerHelper.write(InvokerHelper.java:547)
        at groovy.lang.GString.writeTo(GString.java:194)
        at groovy.lang.GString.toString(GString.java:151)
        at org.codehaus.groovy.runtime.typehandling.ShortTypeHandling.castToString(ShortTypeHandling.java:55)
        at nextflow.trace.StatsObserver.onFlowComplete(StatsObserver.groovy:43)
        at nextflow.Session.shutdown0(Session.groovy:749)
        at nextflow.Session.destroy(Session.groovy:689)
        at nextflow.script.ScriptRunner.terminate(ScriptRunner.groovy:297)
        at nextflow.script.ScriptRunner.execute(ScriptRunner.groovy:134)
        at nextflow.cli.CmdRun.run(CmdRun.groovy:246)
        at nextflow.cli.Launcher.run(Launcher.groovy:451)
        at nextflow.cli.Launcher.main(Launcher.groovy:633)
```

### Program output 

Above is log output.

### Environment 

* Nextflow version: `nextflow version 19.07.0.5106`
* Java version:
```
openjdk version "1.8.0_191"
OpenJDK Runtime Environment (build 1.8.0_191-b12)
OpenJDK 64-Bit Server VM (build 25.191-b12, mixed mode)
```
* Operating system: `Linux`

I've got the same stopping an execution 

```
java.lang.AssertionError: Memory unit cannot be a negative number. Expression: (value >= 0). Values: value = -73014444032
	at org.codehaus.groovy.runtime.InvokerHelper.assertFailed(InvokerHelper.java:417)
	at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.assertFailed(ScriptBytecodeAdapter.java:670)
	at nextflow.util.MemoryUnit.<init>(MemoryUnit.groovy:61)
	at nextflow.util.MemoryUnit.of(MemoryUnit.groovy:172)
	at nextflow.trace.WorkflowStats.toString(WorkflowStats.groovy:255)
	at org.codehaus.groovy.runtime.InvokerHelper.format(InvokerHelper.java:659)
	at org.codehaus.groovy.runtime.InvokerHelper.toString(InvokerHelper.java:148)
	at org.codehaus.groovy.runtime.InvokerHelper.write(InvokerHelper.java:547)
	at groovy.lang.GString.writeTo(GString.java:194)
	at groovy.lang.GString.toString(GString.java:151)
	at org.codehaus.groovy.runtime.typehandling.ShortTypeHandling.castToString(ShortTypeHandling.java:55)
	at nextflow.trace.WorkflowStatsObserver.onFlowComplete(WorkflowStatsObserver.groovy:48)
	at nextflow.Session.notifyFlowComplete(Session.groovy:1008)
	at nextflow.Session.shutdown0(Session.groovy:662)
	at nextflow.Session.destroy(Session.groovy:606)
	at nextflow.script.ScriptRunner.terminate(ScriptRunner.groovy:228)
	at nextflow.script.ScriptRunner.execute(ScriptRunner.groovy:128)
	at nextflow.cli.CmdRun.run(CmdRun.groovy:257)
	at nextflow.cli.Launcher.run(Launcher.groovy:452)
	at nextflow.cli.Launcher.main(Launcher.groovy:634)
```

--- a/modules/nf-commons/src/main/nextflow/util/Duration.groovy
+++ b/modules/nf-commons/src/main/nextflow/util/Duration.groovy
@@ -219,7 +219,8 @@ class Duration implements IDuration, TemporalAmount, Comparable<Duration>, Seria
     }
 
     static Duration between( Temporal start, Temporal end ) {
-        new Duration(java.time.Duration.between(start, end).toMillis())
+        final millis = java.time.Duration.between(start, end).toMillis()
+        new Duration(Math.max(0L, millis))
     }
 
     @Override

--- a/modules/nf-commons/src/test/nextflow/util/DurationTest.groovy
+++ b/modules/nf-commons/src/test/nextflow/util/DurationTest.groovy
@@ -193,10 +193,11 @@ class DurationTest extends Specification {
 
         given:
         def start = Instant.now()
-        def end = start.plusMillis(1000)
 
         expect:
-        Duration.between(start, end) == Duration.of('1sec')
+        Duration.between(start, start.plusMillis(1000)) == Duration.of('1sec')
+        and: 'guard against non-monotonic wall clock'
+        Duration.between(start, start.minusMillis(228)) == Duration.of(0)
     }
 
     // TemporalAmount interface tests

